gridstatus/aeso/aeso.py

```diff
@@ -598,3 +598,180 @@ def get_unit_status(
+            "GAS": "Gas",
+            "COAL": "Coal",
+            "HYDRO": "Hydro",
+            "WIND": "Wind",
+            "SOLAR": "Solar",
+            "ENERGY STORAGE": "Energy Storage",
+            "OTHER": "Biomass and Other",
+        }
+
+        sub_fuel_type_mapping = {
+            "SIMPLE_CYCLE": "Simple Cycle",
+            "COMBINED_CYCLE": "Combined Cycle",
+            "COGENERATION": "Cogeneration",
+            "GAS_FIRED_STEAM": "Gas Fired Steam",
+            "COAL": "Coal",
+            "HYDRO": "Hydro",
+            "OTHER": "Biomass and Other",
+            "ENERGY STORAGE": "Energy Storage",
+            "SOLAR": "Solar",
+            "WIND": "Wind",
+        }
+
+        rows = []
+        for fuel_data in data["return"]:
+            fuel_type = fuel_data["fuel_type"]
+            sub_fuel_type = fuel_data["sub_fuel_type"]
+
+            mapped_fuel_type = fuel_type_mapping.get(fuel_type, fuel_type)
+            mapped_sub_fuel_type = sub_fuel_type_mapping.get(
+                sub_fuel_type,
+                sub_fuel_type,
+            )
+
+            for hour_data in fuel_data["Hours"]:
+                interval_start = pd.to_datetime(
+                    hour_data["begin_datetime_utc"],
+                    utc=True,
+                ).tz_convert(self.default_timezone)
+                interval_end = interval_start + pd.Timedelta(hours=1)
+                outage_grouping = hour_data["outage_grouping"]
+
+                row = {
+                    "Interval Start": interval_start,
+                    "Interval End": interval_end,
+                    "Fuel Type": mapped_fuel_type,
+                    "Sub Fuel Type": mapped_sub_fuel_type,
+                    "Operating Outage": outage_grouping["OP OUT"],
+                    "Mothball Outage": outage_grouping["MBO OUT"],
+                }
+                rows.append(row)
+
+        df = pd.DataFrame(rows)
+
+        current_time = pd.Timestamp.now(tz=self.default_timezone).floor("h")
+        df["Publish Time"] = df.apply(
+            lambda row: (
+                current_time
+                if row["Interval Start"] > current_time
+                else row["Interval Start"] - pd.Timedelta(hours=1)
+            ),
+            axis=1,
+        )
+
+        # NB: Pivot the data to get the by-fuel type outage.
+        df_pivot = df.pivot_table(
+            index=["Interval Start", "Interval End", "Publish Time"],
+            columns=["Sub Fuel Type"],
+            values="Operating Outage",
+            aggfunc="sum",
+        ).reset_index()
+
+        # NB: Sum the operational outage by fuel type to get the total outage.
+        df_pivot["Total Outage"] = df_pivot[
+            [
+                col
+                for col in df_pivot.columns
+                if col
+                not in [
+                    "Interval Start",
+                    "Interval End",
+                    "Publish Time",
+                    "Mothball Outage",
```

Comment:
Mothball outages don't count towards the total outages?

Reply:
I don't really know enough about the details to suggest they should or shouldn't.